OpenRA.Game/Sync.cs

```diff
@@ -62,15 +67,13 @@ static void EmitSyncOpcodes(Type type, ILGenerator il)
 				il.Emit(OpCodes.Ldc_I4, 0x555);
 				il.MarkLabel(l);
 			}
-			else if (type.HasAttribute<SyncAttribute>())
```

Comment:
These particular lines are dead code - we only apply the sync attribute to fields and properties (this is enforced by the attribute usage). In this context `type` will be either a class or struct definition. Therefore, this condition is always false.

I'm not removing the sync attribute or changing how it works - I'm just removing this useless code.